charts/cloudflared/templates/deployment.yaml

```diff
@@ -41,7 +41,7 @@ spec:
             {{- range $key, $val := .Values.ports }}
             {{- if $val.enabled }}
             - name: {{ $key | quote }}
-              containerPort: {{ $val.port }}
+              containerPort: 5053
```

Comment:
Yes you're right, a port change is not working currently. But instead of hardcoding the port value here I would propose to actually change the port inside the container as well.

It seems there is an environment variable available for this: [`TUNNEL_DNS_PORT`](https://github.com/crazy-max/docker-cloudflared#environment-variables)

Best would be to append the variable with the value from `.Values.ports.dns.port` I think.

Do you want to update your PR? Or should I implement this directly? 🙂

Reply:
Propably something like this:

```diff
          env:
            {{- range $i, $val := .Values.env }}
            - name: {{ $val.name | quote }}
              value: {{ $val.value | quote }}
            {{- end }}
+           {{- if and .Values.ports.dns .Values.ports.dns.enabled }}
+           - name: TUNNEL_DNS_PORT
+             value: {{ .Values.ports.dns.port | quote }}
+           {{- end }}
          {{- if and .Values.ports.metrics .Values.ports.metrics.enabled }}
          livenessProbe:
```

Reply:
To be honest, I see no advantage of also changing the port inside the container. As long as the port on the service is not automatically mapped to the containerPort/targetPort, the user can decide which port to use.

But it is your chart. :-) I can of course add this.

Reply:
Unfortunately this does not work, as you cannot bind to port 53 unless the container is running with root permissions:
```
2022-03-17T10:05:34Z INF Starting DNS over HTTPS proxy server address=dns://0.0.0.0:53
2022-03-17T10:05:34Z INF Starting metrics server on [::]:49312/metrics
2022-03-17T10:05:34Z ERR Failed to start the listeners error="failed to create a UDP listener: listen udp 0.0.0.0:53: bind: permission denied"
```

So I would omit that last commit 7040b27 for now, if you are ok with only changing the port on the outside?

Reply:
Yeah this seems legit to me. But the problem with the hardcoded port inside the `range` is that also the metrics port will be hardcoded to `5053`. So let's hardcode both ports inside the pod and completely remove the `range` parts. This should allow to change the ports outside using the service but keep the defaults inside the container. 🙂

Reply:
Ah, I did not have the metrics enabled, so I did not notice.

I'll have a look.